docs/pages/database-access/guides/postgres-redshift.mdx

```diff
@@ -8,253 +8,230 @@ videoBanner: UFhT52d5bYg
+                  "iam:DeleteUserPolicy"
+              ],
+              "Resource": "*"
+          }
+      ]
+  }
+  ```
+  </TabItem>
+  <TabItem label="Redshift / IAM role">
+  Use this policy if your Teleport database agent runs as IAM role (for example,
+  on an EC2 instance with attached IAM role).
+  ```json
+  {
+      "Version": "2012-10-17",
+      "Statement": [
+          {
+              "Effect": "Allow",
+              "Action": [
+                  "redshift:DescribeClusters",
+                  "iam:GetRolePolicy",
+                  "iam:PutRolePolicy",
+                  "iam:DeleteRolePolicy"
+              ],
+              "Resource": "*"
+          }
+      ]
+  }
+  ```
+  </TabItem>
+</Tabs>
+
+### Create IAM permission boundary for Teleport
+Since Teleport will be managing its own IAM policies for access to Redshift
+databases, you need to create a permission boundary to limit its effective
+range of permissions.
+
+Create another managed policy that will serve as a permission boundary on the
+same [Policies](https://console.aws.amazon.com/iamv2/home#/policies) page.
+
+In addition to the set of permissions you created above, the boundary should
+also include `redshift:GetClusterCredentials` which will grant your Teleport
+agent the permssion to generate [temporary
+credentials](https://docs.aws.amazon.com/redshift/latest/mgmt/generating-user-credentials.html)
+to authenticate with Redshift databases.
+
+<Tabs>
+  <TabItem label="Redshift / IAM user">
+  Use this permission boundary if your Teleport database agent runs as IAM user
+  (for example, uses AWS credentials file).
+  ```json
+  {
+      "Version": "2012-10-17",
+      "Statement": [
+          {
+              "Effect": "Allow",
+              "Action": [
+                  "redshift:DescribeClusters",
+                  "redshift:GetClusterCredentials",
+                  "iam:GetUserPolicy",
+                  "iam:PutUserPolicy",
+                  "iam:DeleteUserPolicy"
+              ],
+              "Resource": "*"
+          }
+      ]
+  }
+  ```
+  </TabItem>
+  <TabItem label="Redshift / IAM role">
+  Use this permission boundary if your Teleport database agent runs as IAM role
+  (for example, on an EC2 instance with attached IAM role).
+  ```json
+  {
+      "Version": "2012-10-17",
+      "Statement": [
+          {
+              "Effect": "Allow",
+              "Action": [
+                  "redshift:DescribeClusters",
+                  "redshift:GetClusterCredentials",
+                  "iam:GetRolePolicy",
+                  "iam:PutRolePolicy",
+                  "iam:DeleteRolePolicy"
+              ],
+              "Resource": "*"
+          }
+      ]
+  }
+  ```
+  </TabItem>
+</Tabs>
+
+### Attach policy and boundary to IAM identity
+
+Attach created policy and permission boundary to the IAM identity your Teleport
+database agent will be using. For example, if the agent runs as an IAM user:
```

Comment:
Maybe it's my end-of-the-day brain, but I think we need a sentence explaining what's going on in this image.